source/s3_client.c

```diff
@@ -1130,6 +1189,55 @@ static void s_s3_client_schedule_process_work_synced_default(struct aws_s3_clien
     client->synced_data.process_work_task_scheduled = true;
 }
 
+/* Task function for trying to find a request that can be processed. */
+static void s_s3_client_trim_buffer_pool_task(struct aws_task *task, void *arg, enum aws_task_status task_status) {
+    AWS_PRECONDITION(task);
+    (void)task;
+    (void)task_status;
```

Comment:
debatable: early-out if task_status != RUN?

so we don't have some weird thing happen when this task is canceled during client shutdown?